packages/core/src/renderer/components/cluster-settings/cluster-settings-menu-injection-token.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectionToken } from "@ogre-tools/injectable";
+import type { ClusterPreferences } from "../../../common/cluster-types";
+
+export type ChangedClusterPreference = [keyof ClusterPreferences, any];
```

Comment:
```suggestion
type ClusterPreferenceKeys = keyof ClusterPreferences;
export type ChangedClusterPreference = [ClusterPreferenceKeys, ClusterPreferences[ClusterPreferenceKeys]];
```

That way we will have at least some validation for values.

Reply:
Will check this out.